### PR TITLE
Adding a log for builders found vs build name in presubmit subscription

### PR DESCRIPTION
**Pull request by ricardoamador**
Occasionally we hit an issue where we cannot get an element from the ciYaml and then cannot update an existing task in presubmit. This implies that we found none or more than one task. This log message will let us know which it is if hit again.

*List which issues are fixed by this PR. You must list at least one issue.*
Related to https://github.com/flutter/flutter/issues/135640

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by ricardoamador** [comment]
On a side note, @keyonghan would it be safe to just assume no reschedule and return 1 on this error so we can update the failed task?

File: app_dart/lib/src/request_handlers/presubmit_luci_subscription.dart [hunk 1]
```diff
@@ -119,8 +119,29 @@ class PresubmitLuciSubscription extends SubscriptionHandler {
...
+      final List<String> availableBuilderList = ciYaml.presubmitTargets.map((Target e) => e.value.name).toList();
+      log.warning('ciYaml presubmit targets found: $availableBuilderList');
+      rethrow;
+    }
+
+    // alternative solution, just not reschedule.
+    // This might be allowable. We are checking tip of tree but I don't think the
+    // target has been deleted.
+
+    // if (!ciYaml.presubmitTargets.any((element) => element.value.name == builderName)) {
```
Comment (keyonghan) [inline app_dart/lib/src/request_handlers/presubmit_luci_subscription.dart:137]: Let's go with this solution, and add logs here. This way pub/sub messages will block builds update.

**Comment by keyonghan** [comment]
> On a side note, @keyonghan would it be safe to just assume no reschedule and return 1 on this error so we can update the failed task?

This should be safe. I tried taking a look at the failed case, but didn't find why it complains. With this log, we should get some good idea.

File: app_dart/lib/src/request_handlers/presubmit_luci_subscription.dart (hunk 1 above)
Comment (ricardoamador) [reply app_dart/lib/src/request_handlers/presubmit_luci_subscription.dart:137]: Okay no problem. I will add it.

**Review by keyonghan — COMMENTED** [review]
Could you add a unit test to cover this corner case?

**Review by keyonghan — APPROVED** [review]
LGTM
